libportability-gfx/src/impls.rs

```diff
@@ -1660,23 +1663,28 @@ pub extern "C" fn gfxGetPipelineCacheData(
     pDataSize: *mut usize,
     _pData: *mut ::std::os::raw::c_void,
 ) -> VkResult {
+    //TODO: save
     unsafe { *pDataSize = 0; }
     VkResult::VK_SUCCESS
 }
 #[inline]
 pub extern "C" fn gfxMergePipelineCaches(
-    _device: VkDevice,
-    _dstCache: VkPipelineCache,
-    _srcCacheCount: u32,
-    _pSrcCaches: *const VkPipelineCache,
+    gpu: VkDevice,
+    dstCache: VkPipelineCache,
+    srcCacheCount: u32,
+    pSrcCaches: *const VkPipelineCache,
 ) -> VkResult {
+    let caches = unsafe {
+        slice::from_raw_parts(pSrcCaches, srcCacheCount as usize)
+    };
+    gpu.device.merge_pipeline_caches(&*dstCache, caches.into_iter().map(|h| h.unbox().unwrap()));
```

Comment:
We are effectively destroying/consuming the src caches here, right? I don't think that's the intended design of the vulkan specs.

Reply:
aaah, where have you been before :)
gotta fix gfx-rs first then

Reply:
These iterator interfaces are hard to parse :D